File: PHP/2 SECURITY FUNCTIONS/8 Use of Insufficiently Random Values/php/MOPAS_1_insufficiently_random_values.php
<?php
/**
 *  MOPAS Use of Insufficiently Random Values Example 1
 */
if (empty($_GET['userID'])){
	die('error');
}

$userID = $_GET['userID'];
$sid = generateSessionID($userID);

function generateSessionID($userID){
	srand($userID);
	return rand();
}
?>

File: Java/2 SECURITY FUNCTIONS/8 Use of Insufficiently Random Values/java/MOPAS_1_insufficiently_random_values.java
/**
 *  MOPAS Use of Insufficiently Random Values Example 1
 */
import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.util.Random;

@WebServlet(urlPatterns = {"/MOPAS_1_insufficiently_random_values"})
public class MOPAS_1_insufficiently_random_values extends HttpServlet {

    @Override
    public void doGet(HttpServletRequest request, HttpServletResponse response)
            throws IOException, ServletException {

        String userId = request.getParameter("userId");

        Random ranGen = new Random();
        ranGen.setSeed(Integer.valueOf(userId));

        int sessionId = ranGen.nextInt();
    }
}